### PR TITLE
More automation for release process

### PR DESCRIPTION
**Pull request by X0rg**
Related to #601.

By default, `__version__` is now set to `devel`.
To be brief, `__version__` is set in `dist/copy/xl/version.py` on `make DIST_VERSION=$TRAVIS_TAG dist`.
`__version__` can still be overridden if Exaile is started from source directory.

All tests pass with success.

You can find the dist tarball here: https://github.com/X0rg/exaile/releases/tag/4.0.0-rc4

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,6 +29,28 @@ jobs:
...
+    if: tag IS present
+    install:
+    - sudo apt-get install -y python-gi
+    script:
+    - make DIST_VERSION=$TRAVIS_TAG dist
+    deploy:
+      provider: releases
+      draft: true
+      api_key:
+        secure: "FfKvA/NLFpLLRFWyoDIjPA8CDZ3F7sVDQAxHbRmAFjFtDP2Y0xethxWNisC0BFLlY2NdPVlexlhOYtP87+Z9TCPlAJB/5arabMID34qFxQyeF4Mnuq9xx6YRxlKfY48GWxt7/8pEis9nsH3KzKmoZJl1l7opKXHvccDosE8A/xQiJmWwbGdT40x89CMuOuPbHi/oD/RLcnVjx/O7/KT8PuO2JY4D6bSBZWUuUOtvIv0weMkR/KTz3tNPl+JlVB3n1mSYeBJ+EHIfnbGAc+qzW4Ih31cYLht1hYwiEm0CCGtisCHhRerAgho/QhHPwYN58OHgC60w6GdQk2vMIwyLYzKFL0DjYYmxLs4OXLLJwM1cS79FA9HgSVYc6aDnHVzoPvAOSi0dyhZomuO+IRHYqTO5Q6wN1P63KUN8boq8wYKhA5w8/wWswvBcbhIwsYnfpOhI3wiDfPUbQo8SN9U7hpxcbnQZZ0W+3YwB2dy1KCJ928d0O6Nh+TQrCdsqUUUHIBukPsOJK5cWea8NE76D/77u6CLWMeqOhevAN1PNqvqA5j/Wh5AIzYMJTfqWG74Vq3sOr5YYrCXu80J2Y0XXoW9OKAQUlRxZuypLPK4YKo6oZaGsGeaNKqupXlX62IYCbSNxHKpyIAFmKoEjmgNuJrCtOaTHpVxAsVGQVtsqthw="
```
Comment (X0rg) [inline .travis.yml]: To do.

File: tools/dist.sh [hunk 2]
```diff
@@ -39,12 +40,12 @@ tar --gzip --format=posix --owner 0 --group 0 \
...
 
 #mv tools/installer/exaile-LATEST.exe dist/exaile-${DIST_VERSION}.exe
 
-echo "Successfully built exaile installer! Going to sign the resulting packages"
-echo "with gpg (feel free to CTRL-C at this point if you don't care about that)"
+#echo "Successfully built exaile installer! Going to sign the resulting packages"
+#echo "with gpg (feel free to CTRL-C at this point if you don't care about that)"
 
-gpg --armor --sign --detach-sig dist/exaile-${DIST_VERSION}.tar.gz
+#gpg --armor --sign --detach-sig dist/exaile-${DIST_VERSION}.tar.gz
```
Comment (X0rg) [inline tools/dist.sh]: Maybe you need to configure your key on Travis CI (and on AppVeyor too).

**Comment by virtuald** [comment]
Sorry for the delay, I've been swamped at work all week.

File: .travis.yml (hunk 1 above)
Comment (genodeftest) [reply .travis.yml]: Is this still to do? How about leaving the .travis.yml changes out of this pull request?

Comment (X0rg) [reply .travis.yml]: @genodeftest: Yes, because it is my personal access token.
The `api_key` need to be set with an Exaile access token.

**Comment by virtuald** [comment]
This weekend is the first weekend I've had open in awhile, this is on my calendar.

**Comment by virtuald** [comment]
Thank you so much!

**Comment by X0rg** [comment]
@virtuald: About the `api_key` in `.travis.yml`, I suggest you to take a look about [encryption keys](https://docs.travis-ci.com/user/encryption-keys/) or [environment variables](https://docs.travis-ci.com/user/environment-variables/).

**Comment by virtuald** [comment]
Yeah, doing that now.

**Comment by virtuald** [comment]
Weird, travis deploy works fine, but appveyor seems to break when building a tag now.

**Comment by virtuald** [comment]
I think I figured it out, pushing another test tag. If that doesn't work, will have to wait until tonight.

**Comment by X0rg** [comment]
Fixes in #612.

**Pull request merged**